src/main/java/com/google/cloud/spanner/pgadapter/statements/SessionStatementParser.java

```diff
@@ -279,20 +279,45 @@ static SetStatement parseSetStatement(SimpleParser parser) {
       // Ignore, this is the default.
       parser.eatKeyword("session");
     }
+    if (parser.eatKeyword("time", "zone")) {
```

Comment:
Hmmm... Now that I see this, I think having a completely separate branch for the special `set time zone` is a little too much. Could we keep the if statement here, and set the name as you do here now, and then add an else block that handles reading the variable name and the `to|=` for all other set statements? And then after that keep the parsing of the two statements in a single branch. That means that you also need to set a flag here to remember that it is the special `set time zone` statement, so you can handle the special `local` value. The rest should be the same for both types of statements, as far as I can see, so that should reduce the amount of code duplication.